addons/mail/static/src/core/common/settings_model.js

```diff
@@ -127,27 +128,27 @@ export class Settings extends Record {
     }
 
     /**
-     * @param {string} notif
+     * @param {string} custom_notifications
+     * @param {import("models").Thread} thread
      */
-    setChannelNotifications(notif) {
-        this.store.env.services.orm.call(
-            "res.users.settings",
-            "set_res_users_settings",
-            [[this.id]],
-            {
-                new_settings: {
-                    channel_notifications: notif === "mentions" ? false : notif,
-                },
-            }
-        );
+    async setCustomNotifications(custom_notifications, thread = undefined) {
+        return rpc("/discuss/settings/custom_notifications", {
+            custom_notifications:
+                !thread && custom_notifications === "mentions" ? false : custom_notifications,
+            res_users_settings_id: thread ? undefined : this.id,
+            channel_id: thread?.id,
+        });
```

Comment:
I feel the code would be simpler without passing `res_users_settings_id`.
What I mean is, without passing any `channel_id`, we know this is a global change of discuss notifs, and the actual user is making the request for change so we can deduce the corresponding `res_users_settings`.